### PR TITLE
fix: change zktrie iterator key format

### PR DESCRIPTION
**Pull request by jyc228**
The https://github.com/kroma-network/go-ethereum/issues/71 issue was resolved in the snapsync commit, but it was decided to separate it from snapsync.

## Summary by CodeRabbit

- **New Features**
	- Enhanced data handling in trie structures with the introduction of a new `data` field for improved efficiency and accuracy.
	- Added new functionalities for iterator key management in trie data structures, facilitating better conversion and handling of keys.
	- Introduced new methods for creating and managing tree paths using big integers, expanding the capabilities for tree path manipulation.

### CODE REVIEW

File: trie/zk/tree_path.go [hunk 1]
```diff
@@ -14,6 +16,22 @@ const (
...
 )
 
+func NewTreePathFromHashBig(hash common.Hash) TreePath {
+	return NewTreePathFromBig(hash.Big(), len(hash)*8)
+}
+
+func NewTreePathFromBig(key *big.Int, maxLevel int) TreePath {
+	result := make([]byte, maxLevel)
+	for i := 0; i < maxLevel; i++ {
+		bit := new(big.Int).Mod(key, big.NewInt(2))
```
Comment (Pangssu) [inline trie/zk/tree_path.go:26]:
```suggestion
		bit := new(big.Int).Mod(key, common.Big2)
```

File: trie/iterator.go [hunk 2]
```diff
@@ -858,8 +859,9 @@ func zkMerkleTreeNodeBlobFunctions(findBlobByHash func(key []byte) ([]byte, erro
 			case *zk.LeafNode:
 				return &merkleTreeIteratorLeafNode{
 					hash: hash,
-					blob: n.Data(),
-					key:  zkt.ReverseByteOrder(n.Key),
+					blob: blob,
+					data: n.Data(),
```
Comment (0xHansLee) [inline trie/iterator.go:863]: Just question, is this also required to resolve the related issue?

Comment (jyc228) [reply trie/iterator.go:863]: This is unrelated to the above issue, but should be reflected in snap sync. Should I exclude it?

**Review by Pangssu — APPROVED** [review]

**Review by seolaoh — APPROVED** [review]
LGTM

**Pull request merged**